### PR TITLE
Don't change poetry.lock during `poetry update --dry-run`

### PR DESCRIPTION
**Pull request by martinmo**
# Pull Request Check List

Resolves: #3766

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

AFAIK, there is no relevant documentation to be changed, am I right?

### CODE REVIEW

**Comment by martinmo** [comment]
Linting fails on a file that is not touched in this PR (docs/docs/pyproject.md).

**Comment by abn** [comment]
@martinmo can you rebase the branch please? Should fix the linting issues.

**Review by abn — CHANGES_REQUESTED** [review]
Needs a rebase to fix checks.

**Comment by martinmo** [comment]
> Needs a rebase to fix checks.

Done ✅

**Comment by martinmo** [comment]
I have rebased this to check if it works against the current code in `master`. This PR probably also fixes #3739 (`lock --dry-run`) and #3666 (`remove --dry-run`), or basically every command that has a `--dry-run` flag wrt to the lockfile and doesn't set the `force` parameter explicitely when calling `_write_lock_file(…)`.

**Comment by finswimmer** [comment]
Hello @martinmo,

could you please rebase once more to force the tests running again?

Thanks!

fin swimmer

**Comment by martinmo** [comment]
> Hello @martinmo,
> 
> could you please rebase once more to force the tests running again?
> 
> Thanks!
> 
> fin swimmer

Done ✅

**Comment by martinmo** [comment]
> Hi @martinmo just checking on this and seeing if you had the time to push this through; it's a pretty annoying part of poetry! Thanks :)

Not yet, but I rebased and force pushed it again to see what the CI checks say.

**Comment by martinmo** [comment]
Hi @abn could you please have another look at this PR? I rebased it and adjusted it to conform to the new coding guidelines. Thank you!

**Comment by martinmo** [comment]
@finswimmer the review for this PR is somehow stuck. I saw that you have been active on other PRs in the last days, could you please have a look at this PR? It has a test for the changed behavior and all CI checks have passed. Thanks!

**Comment by dimitrismistriotis** [comment]
Me + team affected from this, would be great if it could be progressed.

**Comment by zEdS15B3GCwq** [comment]
I'm on 1.2.0b1 (current latest) and this issue persist. Please proceed with the PR. Much appreciated.

File: tests/console/commands/test_update.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    pyproject_content = (source / "pyproject.toml").read_text(encoding="utf-8")
+    poetry_lock_content = (source / "poetry.lock").read_text(encoding="utf-8")
+    return project_factory(
+        name="foobar",
+        pyproject_content=pyproject_content,
+        poetry_lock_content=poetry_lock_content,
+    )
+
+
+def test_dry_run_update(
```
Comment (radoering) [inline tests/console/commands/test_update.py:37]: Can you please add similar tests (or extend existing tests) for the following commands (if not yet existing):

- `add --dry-run`
- `add --lock --dry-run`
- `remove --dry-run`

Further, it seems that tests for `add` check that the `pyproject.toml` is not changed for dry-run. Can you please add this check here, too?

Comment (radoering) [reply tests/console/commands/test_update.py:37]: And probably, you should rebase first before adding tests. 🙏

**Pull request closed without merging**